### PR TITLE
restore invisible terms / privacy pages

### PR DESCRIPTION
**Pull request by 6a68**
## Because

- The terms and privacy pages aren't visible if they are visited via the link on the signin form (so that they are templated by the front-end code)

Pulling over my description of the bug from a comment on that issue:

> It turns out there used to be two versions of the terms and privacy notices: one rendered as a partial template, and one rendered as a full html page. The partials were lost in some recent changes, so when the page is served up, a page is loaded into the bigger page. The terms / privacy pages appear blank because there's a fade-in effect applied to the content wrapper, `<div id="stage">`, but now that an html page is getting loaded into another page, there are actually two nested content wrapper els with that ID, and the fade-in animation is only getting applied to the outer one:
> 
> Screen Shot 2021-02-25 at 3 53 55 PM
> 
> The fix is to update the l10n generator grunttasks to restore the missing partial templates.

## This pull request

- This commit builds on some recent l10n changes to fix this bug:

* restore the partial templates of the translated legal docs and, following the old approach, store them in /privacy and /terms directories under server/templates/pages/dist/.

* update the legal-templates lib to load the partials correctly

* Big open question: if I set defaultLegalLang to en, something breaks in the grunttasks. Are we okay with the default legal language being en_US, or does it need to be en?

For testing locally:
* the partial should render correctly if you navigate from the signin form to the terms or privacy pages (by clicking the link in the email input form).
* the full page should render correctly If you go to the /terms or /privacy pages directly, or if you follow the link from the signin form, then hard refresh the page (which triggers a server render).

## Issue that this pull request solves

Closes: #7516

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

**Comment by 6a68** [comment]
Hey @meandavejustice, I know you're busy with other stuff, but some of these changes build on the l10n changes you landed recently. Mind taking a quick look at this draft, just to call out anything important I might have missed (or broken :-P)?

File: packages/fxa-content-server/grunttasks/remarkable.js [hunk 1]
```diff
@@ -12,15 +12,9 @@ module.exports = function (grunt) {
...
 
-    if (destFile.includes('privacy')) {
-      return path.join(destPath, lang, 'privacy.html');
-    } else if (destFile.includes('tos')) {
-      return path.join(destPath, lang, 'terms.html');
-    } else {
-      return path.join(destPath, i18n.localeFrom(destFile) + '.html');
-    }
+    const lang = i18n.localeFrom(destFile.split('/')[0]);
+    return path.join(destPath, lang + '.html');
```
Comment (6a68) [inline packages/fxa-content-server/grunttasks/remarkable.js:17]: for example, the path is something like `/server/templates/pages/dist/privacy/en_US.html`

File: packages/fxa-content-server/grunttasks/yeoman.js [hunk 2]
```diff
@@ -16,12 +16,14 @@ module.exports = function (grunt) {
     es5: '.es5',
     page_template_dist: TEMPLATE_ROOT + '/pages/dist',
     page_template_src: TEMPLATE_ROOT + '/pages/src',
+    pp_html_dest: TEMPLATE_ROOT + '/pages/dist/privacy',
```
Comment (6a68) [inline packages/fxa-content-server/grunttasks/yeoman.js:19]: this and tos_html_dest are needed to put the partials in the correct locations

File: packages/fxa-content-server/server/lib/legal-templates.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ module.exports = function (i18n, root) {
 
     // Filenames are normalized to locale, not language.
     const locale = i18n.localeFrom(lang);
-    const templatePath = path.join(root, locale, type + '.html');
+    const templatePath = path.join(root, type, locale + '.html');
```
Comment (6a68) [inline packages/fxa-content-server/server/lib/legal-templates.js:17]: This matches the path in the grunttasks/remarkable diff above

**Review by dannycoates — APPROVED** [review]
lgtm if we fix the failing tests

**Comment by 6a68** [comment]
> lgtm if we fix the failing tests

s/en_US/en-US 🙄

**Pull request merged**